src/main/java/com/google/devtools/build/lib/packages/ConfiguredAttributeMapper.java

```diff
@@ -220,11 +225,50 @@ public void validateAttributes() throws EvalException {
     return matchingResult;
   }
 
+  /**
+   * Constructs a <a href="https://bazel.build/designs/2016/05/23/beautiful-error-messages.html">
+   * beautiful error</a> for when no conditions in a configurable attribute match.
+   */
+  private static String noMatchError(
+      String attribute,
+      String customNoMatchError,
+      LinkedHashSet<Label> conditionLabels,
+      Label targetLabel,
+      String configHash) {
+    String error =
+        String.format(
+            "configurable attribute \"%s\" in %s doesn't match this configuration",
+            attribute, targetLabel);
+    if (!customNoMatchError.isEmpty()) {
+      error += String.format(": %s\n", customNoMatchError);
+    } else {
+      error +=
+          ". Would a default condition help?\n\n"
+              + "Conditions checked:\n "
+              + Joiner.on("\n ").join(conditionLabels)
+              + "\n\n"
+              + "To see a condition's definition, run: bazel query --output=build "
+              + "<condition label>.\n";
+    }
+    // See ConfiguredTargetQueryEnvironment#shortID for the substring rationale.
+    String configShortHash = configHash.substring(0, 7);
+    error +=
+        String.format(
+            "\nThis instance of %s has configuration identifier %s. "
+                + "To inspect its configuration, run: bazel config %s.\n",
+            targetLabel, configShortHash, configShortHash);
+    error +=
+        "\n"
+            + "For more help, see"
+            + " https://docs.bazel.build/configurable-attributes.html#why-doesnt-my-select-choose-what-i-expect.\n\n";
+    return error;
+  }
+
   @Override
   public <T> T get(String attributeName, Type<T> type) {
     try {
       return getAndValidate(attributeName, type);
-    } catch (EvalException e) {
+    } catch (ValidationException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)